test/jdk/java/lang/ProcessBuilder/JspawnhelperProtocol.java

```diff
@@ -92,6 +93,40 @@ private static void normalExec() throws Exception {
         }
     }
 
+    private static void jspawnhelperWithNoArgs() throws Exception {
+        System.out.println("Running jspawnhelper without args");
+        Process p = null;
+        try {
+            p = Runtime.getRuntime().exec(Paths.get(System.getProperty("java.home"), "lib", "jspawnhelper").toString());
+        } catch (Exception e) {
+            e.printStackTrace(System.out);
+            System.exit(ERROR);
+        }
+        if (!p.waitFor(TIMEOUT, TimeUnit.SECONDS)) {
+            System.out.println("Child process timed out");
+            System.exit(ERROR + 1);
+        }
+        if (p.exitValue() != 1)
+            System.exit(ERROR + 2);
+        System.exit(0);
```

Comment:
Its bad form to System.exit from not at the top level. Is that necessary.

Reply:
This is in line with other tests in this file , e.g. see `parentCode()`. 

I agree that code would be cleaner if the test is refactored to return error code from the test methods and system.exit() in the main instead, but this might be a separate pr done before or after this one.

Reply:
ok, the same pattern is used consistently.